### ISSUE TITLE
Are the heat source capacities properly scaled?

### ISSUE THREAD

**Issue by ChaelKruip** [quintel/etmoses#1231]
Looking at the heat chart I see that geothermal only produces 370.6 kW (I think, as the axes has no label):
![etmoses](https://cloud.githubusercontent.com/assets/1303760/16686110/1aca28ce-4510-11e6-8863-30f9036d2c50.png)

In the source table I have installed 100 units of 3.7 MW = 370 MW:
![etmoses](https://cloud.githubusercontent.com/assets/1303760/16686134/4ac26fe6-4510-11e6-9e6c-0088a9d12d48.png)

Is there a factor 1000 not taken into account?

**Comment by antw** [comment]
> Is there a factor 1000 not taken into account?

Indeed. I'll fix it...

**Comment by antw** [comment]
Is there a reason why we use MW there when the rest of the application uses kW?

The unit is also incorrect for costs which all specify `kEUR` but in fact the business case shows that the numbers are just being treated as `EUR`. I have a pipe with investment costs of €1,200,000, lifetime of 50 years, and O&M costs of €12,000. In the business case, the price shown is €36,000.

Was there a decision made to use MW and kEUR instead of kW and EUR? If not, can I change it?

**Comment by ChaelKruip** [comment]
> Is there a reason why we use MW there when the rest of the application uses kW?

I think the numbers would be too big to display conveniently in kW, so the table shows it in MW. Perhaps that is why it is stored in MW as well?

>  If not, can I change it?

Yes please!

**Comment by antw** [comment]
> Yes please!

Will do.

The must-run production value appears to also be incorrect. It takes the value in PJ [from the query](https://github.com/quintel/etsource/blob/master/gqueries/modules/etmoses/etmoses_must_run_heat_for_households_from_central_heat_network.gql) (the query unit is specified as MJ but [the "upstream" query returns PJ](https://github.com/quintel/etsource/blob/master/gqueries/output_elements/output_series/mekko_174_heat_network/households_heat_network_connection_steam_hot_water_in_heat_network_mekko.gql)) without converting to Wh.

**Comment by ChaelKruip** [comment]
> The must-run production value appears to also be incorrect. It takes the value in PJ from the query (the query unit is specified as MJ but the "upstream" query returns PJ) without converting to Wh.

Well spotted! Shall I fix these queries? What unit would the ETE output be in ideally?

**Comment by antw** [comment]
I suppose that [the Moses query](https://github.com/quintel/etsource/blob/master/gqueries/modules/etmoses/etmoses_must_run_heat_for_households_from_central_heat_network.gql) should either specify the unit as `PJ`, or multiply the value by a billion to convert back to `MJ`. My preference would be for the latter, but I don't mind either way.

**Comment by ChaelKruip** [comment]
> My preference would be for the latter, but I don't mind either way.

Done! https://github.com/quintel/etsource/commit/8e032f5ada9310302116547d9a519b85e2154c0b

**Comment by antw** [comment]
I have updated the units used in the heat source lists, fixed the import attribute classes to convert to the new units, and migrated old LESes.

1. Must-run central heat was importing heat demand from ETEngine as MJ (as is standard) but did not convert that value to kWh.

2. Some prices were labelled with an absurd kEUR unit (kEUR/kW would be better expressed as EUR/MW, and kEUR/MW as EUR/GW). Despite this, at no point was the price then multiplied by 1000 in the business case, to get back to euros. Therefore I believe prices calculated for the market model were wrong.

  I have changed all units on the heat source form to use "EUR", "kW", and "kWh". It may be that these are better expressed to visitors as "MW" and "MWh", but the model layer and database should *not* be storing them as such.

3. Prices of heat assets were expressed in kEUR, and I'm skeptical of their accuracy. I find it hard to believe that 1km of "standard pipe" costs 1,2000,000 kEUR (1.2 billion euros) to install and a "city apartment" connection costs 2.2 million:

  <img width="441" alt="screen shot 2016-07-09 at 15 44 58" src="https://cloud.githubusercontent.com/assets/4383/16708448/2725c13c-45ec-11e6-9ea5-e87084aee854.png">

  I'm assuming that those were incorrectly labelled and that the real unit is also EUR. Either way, those prices were also not multiplied by 1000, so the values were treated as euros anyway.

4. Costs for the central heat network were described as EUR/MW so I changed them to EUR/kW. However this makes the network very cheap to operate relative to the other technologies; I would appreciate a second opinion on whether this is correct:

  <img width="918" alt="screen shot 2016-07-09 at 15 55 54" src="https://cloud.githubusercontent.com/assets/4383/16708491/b336a514-45ed-11e6-9c0b-92e23c205465.png">

--

**For the future, some ground-rules:**

* Applications **must** use a common unit internally. *It is not okay* to mix-and-match kW and MW, or EUR and kEUR. It is inevitable that one day a value in kW is going to be accidentally treated as MW, or vice-versa (as point 2 shows above), and I spent a lot of time trying to keep track of which variables were expressed in which units. ETEngine uses MJ *everywhere* to describe energy, and therefore this mistake doesn't happen there.

  ETMoses has so far used kWh, therefore every variable or database value which stores a value for energy **should do so using kWh as the unit**.

  For reference: 1Wh = 3600J.

* If it is necessary to show the user a value in a different unit (showing 120,000 kWh as 120 MWh) **then that is a presentation detail**. The raw kWh value should be converted to the new unit in the presentation layer (a Rails helper or view, or by JavaScript) and converted back to the original unit *before* it being handed back to the Model layer.

  For example, in the heat source form you might present "120,000 kWh" as "120 MWh". When the user saves that form, the value must be converted back to the kWh representation by JavaScript, the Rails controller, or a `SomethingUpdater` helper class which sits between the controller and model (in fact, [this is exactly what service objects are supposed to be about](https://www.netguru.co/blog/service-objects-in-rails-will-help)). Either way, the model and database should never see a value expressed in MWh.

* Forms must use a common unit. If one field shows energy in MWh, *all* fields describing energy must do so.

  <img width="304" alt="screen shot 2016-07-09 at 16 13 32" src="https://cloud.githubusercontent.com/assets/4383/16708567/656b27c6-45f0-11e6-9113-d7be56606c8c.png">

--

*(pinging @grdw and @dennisschoenmakers)*
